src/Components/News/NewsCreateEdit.js

```diff
@@ -0,0 +1,231 @@
+          <img
+            src={file.preview}
+            style={img}
+            alt='img not found'
+          />
+        </div>
+      </div>
+    ));
+  
+    useEffect(() => () => {
+      // Make sure to revoke the data uris to avoid memory leaks
+      files.forEach(file => URL.revokeObjectURL(file.preview));
+    }, [files]);
+
+    const handleChange = (e) => { //actualiza el estado de title y category
+        if (e.target.name === 'title') {
+            setInitialValues({ ...initialValues, title: e.target.value })
+        } if (e.target.name === 'category') {
+            setInitialValues({ ...initialValues, category: e.target.value })
+        }
+    }
+
+   
+    const contentChange = (e, editor) => { //actualiza el estado de content
+        const data = editor.getData()
+        setInitialValues({
+            ...initialValues, content: data
+        })
+    }
+
+    const handleSubmit = (e) => { //envia los datos dependiendo del id hace una peticion post o put
+        const datosConEtiquetas = initialValues.content
+        const datosSinEtiquetas = datosConEtiquetas.replace(/<[^>]+>/g, ''); //convierte a texto plano el content
+
+        if (!id) {
+            axios.post(urlCrearNovedad, {
+                name: initialValues.title,
+                image: initialValues.image,
+                content: datosSinEtiquetas,
+                category_id: initialValues.category
+            }).then(res => {
+                if (res.status === 200) {
+                    alert("news create successfulli");
+                    return setInitialValues(datosForm)
+                }
+            }).catch(err => {
+                alert(err)
+            })
+        }
+        if (id) {
+            axios.put(urlEditarNovedad, {
+                name: initialValues.title,
+                image: initialValues.image,
+                content: datosSinEtiquetas,
+                category_id: initialValues.category
+            }).then(res => {
+                if (res.status === 200) {
+                    alert("news updated successfulli");
+                    return setInitialValues(datosForm)
+                }
+            }).catch(err => {
+                alert(err)
+            })
+        }
+        e.preventDefault();
+        console.log(initialValues);
+    }
+
+    useEffect(() => {
+
+        const DatosEditNew = async () => {
+            const DataInicialCategoria = await axios.get(urlCategoria),
+                categoriaData = await DataInicialCategoria.data.data;
+            setcategorias(categoriaData)
+
+            if (id) {
+                const datosIniciales = await axios.get(urlEditarNovedad),
+                    EditNewData = datosIniciales.data.data,
+                    { name, content } = await EditNewData
+                if (name) setInitialValues({ ...initialValues, title: name })
+                if (content) setInitialValues({ ...initialValues, content: content })
+            }
+
+        }
```

Comment:
Please use english and do not mix languages

Reply:
delete extra spaces

Reply:
The name of this function do not reflect what this function do and the code is too mixed, please move the axios call to the respective service. Then create a function out of the useEffect that update the objects

Reply:
The creation of this function should be out of the useEffect